www/class/centreonACL.class.php

```diff
@@ -439,6 +439,95 @@ private function setTopology()
                 }
             }
         }
+        $this->checkTopology();
+    }
+    
+    /**
+     * Use to check and fix if in the topology, a parent has access rights that
+     * can be higher than children when they have the same endpoint.
+     */
+    private function checkTopology(): void
+    {
+        if (!empty($this->topology)) {
+            /**
+             * Filter to keep the first child available per level.
+             */
+            $getFirstChildPerLvl = function (array $topologies): ?array {
```

Comment:
It seems that there is no case where you return "null".
So return type should be "array" instead of "?array"